### PR TITLE
chore: upgrade to deno 1.16

### PR DESCRIPTION
**Pull request by JohannesRudolph**
this contains a breaking change in the unstable signal handler API

### CODE REVIEW

File: src/process/loader-shell-runner.test.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Deno.test("unregistering a signal handler that was never registered has no bad side-effects", () => {
+  Deno.removeSignalListener("SIGTERM", () => {});
```
Comment (tfelix) [inline src/process/loader-shell-runner.test.ts:2]: d: I guess you dont need to test for any conditions here? Just remove it and that is basically the test?

File: src/process/loader-shell-runner.ts [hunk 2]
```diff
@@ -76,24 +72,15 @@ export class LoaderShellRunner implements IShellRunner {
 
   private hideCursor() {
     // Setup a watch for interrupt signals to display the cursor again in case of SIGINT or SIGTERM
-    this.sigInt = Deno.signal("SIGINT");
-    this.sigInt!.then(() => this.forceStopLoading());
-
-    this.sigTerm = Deno.signal("SIGTERM");
-    this.sigTerm!.then(() => this.forceStopLoading());
+    Deno.addSignalListener("SIGINT", this.sigIntHandler);
```
Comment (tfelix) [inline src/process/loader-shell-runner.ts:75]: looks a lot nicer now

**Review by tfelix — APPROVED** [review]
lgtm

File: src/process/loader-shell-runner.test.ts (hunk 1 above)
Comment (JohannesRudolph) [reply src/process/loader-shell-runner.test.ts:2]: yip

**Pull request merged**